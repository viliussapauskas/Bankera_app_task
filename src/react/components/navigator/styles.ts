import { NativeStackNavigationOptions } from '@react-navigation/native-stack'
import { StyleSheet } from 'react-native'

export const styles = StyleSheet.create({
    header: {
        flex: 1,
        padding: 30,
        backgroundColor: '#fff',
    },
})

export const screenOptions: NativeStackNavigationOptions = {
    headerStyle: {
        backgroundColor: '#f4511e',
    },
    headerTintColor: '#fff',
    headerTitleStyle: {
        fontWeight: 'bold',
    },
    headerBackTitleStyle: {},
}
