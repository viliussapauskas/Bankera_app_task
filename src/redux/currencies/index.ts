export { fetchCurrencies } from './currenciesAPI'
export {
    currenciesReducer,
    currenciesActionCreators,
    currenciesSlice,
    getCurrenciesStateSelector,
    loadCurrenciesAsync,
} from './currenciesSlice'
export { initialState } from './state'
export { type Currency } from './types'
