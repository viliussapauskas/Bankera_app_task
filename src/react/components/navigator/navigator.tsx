import React from 'react'
import { NavigationContainer } from '@react-navigation/native'
import { createNativeStackNavigator } from '@react-navigation/native-stack'
import { CurrencyScreen } from '../../currencyScreen'
import { WalletScreen } from '../../walletScreen'
import { screenOptions } from './styles'

const Stack = createNativeStackNavigator()

export const Navigator = () => {
    return (
        <>
            <NavigationContainer>
                <Stack.Navigator screenOptions={{ ...screenOptions }}>
                    <Stack.Screen
                        name="Wallet"
                        component={WalletScreen}
                        options={{ title: 'Your wallet' }}
                    />
                    <Stack.Screen name="Currency" component={CurrencyScreen} />
                </Stack.Navigator>
            </NavigationContainer>
        </>
    )
}
