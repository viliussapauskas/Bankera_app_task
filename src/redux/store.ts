import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit'
import { currenciesReducer } from './currencies'

export const store = configureStore({
    reducer: {
        currencies: currenciesReducer,
    },
})

export type AppDispatch = typeof store.dispatch
export type RootState = ReturnType<typeof store.getState>
export type AppThunk<ReturnType = void> = ThunkAction<
    ReturnType,
    RootState,
    unknown,
    Action<string>
>
