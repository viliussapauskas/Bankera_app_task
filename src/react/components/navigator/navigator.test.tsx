import { render } from '@testing-library/react-native'
import { Provider } from 'react-redux'
import { store } from '../../../redux/store'
import { Navigator } from './navigator'

describe('Navigator component tests', () => {
    it('Should render Navigator component', () => {
        const component = render(
            <Provider store={store}>
                <Navigator />
            </Provider>
        )
        expect(component.toJSON()).toMatchSnapshot()
    })
})
