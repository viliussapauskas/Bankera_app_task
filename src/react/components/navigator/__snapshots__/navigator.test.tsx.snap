// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigator component tests Should render Navigator component 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="#f4511e"
        color="#fff"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Your wallet"
        titleColor="#fff"
        titleFontWeight="bold"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <RCTSafeAreaView
          emulateUnlessSupported={true}
          style={
            Object {
              "flex": 1,
            }
          }
          testID="layout"
        >
          <RCTScrollView
            accessible={false}
            contentContainerStyle={
              Object {
                "flex": 1,
                "padding": 20,
              }
            }
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            refreshControl={
              <RefreshControlMock
                onRefresh={[Function]}
                refreshing={false}
              />
            }
          >
            <RCTRefreshControl />
            <View>
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
                testID="walletScreen"
              >
                <View
                  accessibilityRole="progressbar"
                  accessibilityState={
                    Object {
                      "busy": true,
                    }
                  }
                  accessible={true}
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                  testID="spinner"
                >
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "height": 48,
                        "opacity": 1,
                        "width": 48,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        Object {
                          "alignItems": "center",
                          "bottom": 0,
                          "justifyContent": "center",
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        style={
                          Object {
                            "height": 48,
                            "transform": Array [
                              Object {
                                "rotate": "45deg",
                              },
                            ],
                            "width": 48,
                          }
                        }
                      >
                        <View
                          collapsable={false}
                          style={
                            Object {
                              "height": 24,
                              "overflow": "hidden",
                              "width": 48,
                            }
                          }
                        >
                          <View
                            collapsable={false}
                            style={
                              Object {
                                "height": 48,
                                "transform": Array [
                                  Object {
                                    "translateY": 0,
                                  },
                                  Object {
                                    "rotate": "-165deg",
                                  },
                                ],
                                "width": 48,
                              }
                            }
                          >
                            <View
                              collapsable={false}
                              style={
                                Object {
                                  "height": 24,
                                  "overflow": "hidden",
                                  "width": 48,
                                }
                              }
                            >
                              <View
                                collapsable={false}
                                style={
                                  Object {
                                    "borderColor": "#f4511e",
                                    "borderRadius": 24,
                                    "borderWidth": 4.8,
                                    "height": 48,
                                    "width": 48,
                                  }
                                }
                              />
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                    <View
                      collapsable={false}
                      style={
                        Object {
                          "alignItems": "center",
                          "bottom": 0,
                          "justifyContent": "center",
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        style={
                          Object {
                            "height": 48,
                            "transform": Array [
                              Object {
                                "rotate": "45deg",
                              },
                            ],
                            "width": 48,
                          }
                        }
                      >
                        <View
                          collapsable={false}
                          style={
                            Object {
                              "height": 24,
                              "overflow": "hidden",
                              "top": 24,
                              "width": 48,
                            }
                          }
                        >
                          <View
                            collapsable={false}
                            style={
                              Object {
                                "height": 48,
                                "transform": Array [
                                  Object {
                                    "translateY": -24,
                                  },
                                  Object {
                                    "rotate": "345deg",
                                  },
                                ],
                                "width": 48,
                              }
                            }
                          >
                            <View
                              collapsable={false}
                              style={
                                Object {
                                  "height": 24,
                                  "overflow": "hidden",
                                  "width": 48,
                                }
                              }
                            >
                              <View
                                collapsable={false}
                                style={
                                  Object {
                                    "borderColor": "#f4511e",
                                    "borderRadius": 24,
                                    "borderWidth": 4.8,
                                    "height": 48,
                                    "width": 48,
                                  }
                                }
                              />
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </RCTScrollView>
        </RCTSafeAreaView>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
